sdk/core/azure-core/inc/azure/core/internal/http/pipeline.hpp

```diff
@@ -145,6 +145,88 @@ namespace Azure { namespace Core { namespace Http { namespace _internal {
               clientOptions.Transport));
     }
 
+    /**
+     * @brief Construct a new HTTP Pipeline object from clientOptions.
+     *
+     * @remark The client options includes per retry and per call policies which are merged with the
+     * service-specific per retry policies.
+     *
+     * @remark This specialization of the HttpPipeline constructor constructs an HTTP pipeline
+     * *without* a telemetry policy. It is intended for use by service clients which have converted
+     * to use distributed tracing - the distributed tracing policy adds the User-Agent header to the
+     * request.
+     *
+     * @param clientOptions The SDK client options.
+     * @param perRetryPolicies The service-specific per retry policies.
+     * @param perCallPolicies The service-specific per call policies.
+     */
+    explicit HttpPipeline(
```

Comment:
Can you write it in a way that that delegates both constructors to one function, possibly a private constructor, or an init function that initializes m_policies, so that there is minimum code repetition.